packages/remix-dev/compiler.ts

```diff
@@ -425,6 +441,11 @@ function createServerBuild(
     plugins.unshift(NodeModulesPolyfillPlugin());
   }
 
+  let yarnPnpPlugin = await getYarnPnpPlugin();
+  if (yarnPnpPlugin) {
+    plugins.push(yarnPnpPlugin);
+  }
```

Comment:
I'm trying to determine why we are adding the Yarn PnP plugin conditionally. Is it because the plugin wouldn't work with users using `npm` or `yarn`? Or is it for having `@yarnpkg/esbuild-plugin-pnp` in `peerDependencies` and cut down the number of dependencies for non-Yarn-PnP users?

Reply:
Yep, it's an optional peer dependency, so for most users there is no point in having it. Yarn PNP users would have to explicitly install it themselves.